packages/discord.js/src/structures/interfaces/TextBasedChannel.js

```diff
@@ -165,6 +165,11 @@ class TextBasedChannel {
       return dm.send(options);
     }
 
+    if (typeof options.nonce === 'undefined' && this.client.options.enforceNonce) {
```

Comment:
There is no `undefined` type, you should check if the typeof is object and then if nonce isnt undefined